crates/primitives/src/net.rs

```diff
@@ -66,17 +71,46 @@ pub fn parse_nodes(nodes: impl IntoIterator<Item = impl AsRef<str>>) -> Vec<Node
     nodes.into_iter().map(|s| s.as_ref().parse().unwrap()).collect()
 }
 
+/// Error thrown when network can't be identified on a node record.
+#[non_exhaustive]
+#[derive(Debug, Error)]
+pub enum NetworkIdError {
+    /// Missing key used to identify an execution layer enr on Ethereum network.
+    #[error("fork id missing on enr, 'eth' key missing")]
+    EthForkIdMissing,
+    /// Failed to decode fork ID rlp value.
+    #[error("failed to decode fork id, 'eth': {0:?}")]
+    ForkIdDecodeError(Vec<u8>),
+}
+
+/// Tries to read the [`ForkId`] from given [`Enr`].
+pub fn get_fork_id(enr: &Enr<secp256k1::SecretKey>) -> Result<ForkId, NetworkIdError> {
+    let Some(mut maybe_fork_id) = enr.get(b"eth") else {
+        return Err(NetworkIdError::EthForkIdMissing)
+    };
+
+    let Ok(fork_id) = ForkId::decode(&mut maybe_fork_id) else {
+        return Err(NetworkIdError::ForkIdDecodeError(maybe_fork_id.to_vec()))
+    };
+
+    Ok(fork_id)
+}
```

Comment:
do we need this?

I'm skeptical that this will be useful.
looks like this is unused here, so I'd like to exclude this from this pr and reevaluate once this becomes useful

Reply:
the whole conversion to `NodeRecord` from `Enr` via `TryFrom` impl isn't used in this pr after addressing review, not only `get_fork_id`. can cherry-pick to other pr that would like to use these methods.